### PR TITLE
Fixes null.has_reagent() runtimes caused by on_mob_life procs.

### PR DESCRIPTION
**Pull request by gbasood**
Also absolute paths Chemistry-Reagents because I did it anyways when I was trying to get stack dumps working
And (doubtfully) fixes a few qdeletions not being properly handled.
#### Why this happened

> <Sood> during /datum/reagent/proc/on_mob_life, some procs are called which can result in the reagent's deletion
> <Sood> er
> <Sood> qdel'tion anyway
> <Sood> but even if that occurs the reagent keeps on processing it's overriden proc
> <Sood> for example, dexalin
> <Sood> it's on_mob_life calls it's parent on_mob_life. when its volume is <0.1 it gets qdel'd in there. and then it continues with its own custom on_mob_life
> <Sood> which also explains why each reagent only runtimes once because it's been properly removed from the reagents list and isn't called again on the next loop through

Or, put another way:
- dexalin's on_mob_life is called
- it calls ..() (/datum/reagent/proc/on_mob_life())
- somewhere in there it checks the reagent's volume. if that volume < 0.1 it qdel's the reagent.
- the ..() call continues. it does not check if the holder still exists before the proc ends.
- execution re-enters the overriden dexalin on_mob_life() proc
- it tries to access its holder var, which is now nulled out because Destroy() has been called on it
- RUNTIME
#### What I did:

In /datum/reagents/proc/on_mob_life(), it now checks if the reagent's holder var is anything other than null before continuing processing. If it is null, it returns 1. All the overriden on_mob_life()'s (that I checked) checked if ..() returned 1 and then stopped processing before continuing. So it never gets to the part that it was runtiming at before.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -383,6 +385,7 @@
 		lowertemp.react()
 		T.assume_air(lowertemp)
 		qdel(hotspot)
+		hotspot = null
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: Temporary variables like this dont need to be nulled

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -1589,6 +1601,7 @@
 			for(var/obj/effect/E in W)
 				if(E.name == "Wallrot")
 					qdel(E)
+					E = null
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: I just. What.

**Comment by gbasood** [comment]
okay i'm not nulling temp vars anymore are you happy

**Comment by clusterfack** [comment]
Yes you are, you are nulling references that **do not need to be nulled**

A variable in a for loop
A variable in a proc

Neither of these need to be nulled

**Comment by gbasood** [comment]
okay but those two things are the only things I actually changed as far as nulling things goes in this PR
I removed them in that last commit
EDIT: I found a thing that while I may not actually be at fault for, is still bad, hold on let me go fix it

**Comment by gbasood** [comment]
so on line 609 of Chemistry-Reagents.dm, dudes who got mutated into slimes and had implants in them would have their implants qdel'd but not have the implant removed from their contents

not super sure that matters for qdel'ing purposes, and especially since this happens so little in a round it probably isn't important, but I changed the code to remove it from the mob before qdel'ing it

**Comment by gbasood** [comment]
nevermind I'm dumb
I squashed shit and made sure I wasn't qdel'ing anything that didn't need it

**Pull request merged**